aten/src/ATen/native/cuda/Reduce.cuh

```diff
@@ -508,8 +538,38 @@ static void launch_reduce_kernel(const ReduceConfig& config, const R& reduction)
   AT_CUDA_CHECK(cudaGetLastError());
 }
 
+struct AccumulationBuffer {
+  AccumulationBuffer() {}
+
+  AccumulationBuffer(size_t acc_stride, size_t out_stride, char* out_ptr, int64_t size) {
```

Comment:
acc_stride and out_stride names are misleading, they are element sizes?

Reply:
they are sizeof(acc_t) & sizeof(out_t)
I'll rename them to acc_t_size, out_t_size.